packages/dds/tree/src/core/edit-manager/editManager.ts

```diff
@@ -287,16 +290,76 @@ export class EditManager<
 		return this.changeFamily.intoDelta(this.rebaseLocalBranchOverTrunk());
 	}
 
-	public addLocalChange(revision: RevisionTag, change: TChangeset): Delta.Root {
+	public addLocalChange(
+		revision: RevisionTag,
+		change: TChangeset,
+		rebaseAnchors = true,
+	): Delta.Root {
 		this.pushToLocalBranch(revision, change);
 
-		if (this.anchors !== undefined) {
+		if (rebaseAnchors && this.anchors !== undefined) {
 			this.changeFamily.rebaser.rebaseAnchors(this.anchors, change);
 		}
 
 		return this.changeFamily.intoDelta(change);
 	}
 
+	/**
+	 * Given a revision on the local branch, replace all commits after it with a single commit containing
+	 * an equivalent composition of changes.
+	 * @returns the new commit (which is now the head of the local branch)
+	 */
+	public squashLocalChanges(startRevision: RevisionTag): Commit<TChangeset> {
+		const [squashStart, commits] = this.findLocalCommit(startRevision);
+		// Anonymize the commits from this transaction by stripping their revision tags.
+		// Otherwise, the change rebaser will record their tags and those tags no longer exist.
+		const anonymousCommits = commits.map(({ change }) => ({ change, revision: undefined }));
+
+		{
+			const change = this.changeFamily.rebaser.compose(anonymousCommits);
+			this.localBranch = mintCommit(squashStart, {
+				revision: mintRevisionTag(),
+				sessionId: this.localSessionId,
+				change,
+			});
+			return this.localBranch;
+		}
+	}
+
+	/**
+	 * Given a revision on the local branch, remove all commits after it.
+	 * @param repairStore - an optional repair data store to assist with generating inverses of the removed commits
+	 * @returns an iterator of deltas where each delta describes the change from rolling back one of the removed commits.
+	 */
+	public *rollbackLocalChanges(
+		startRevision: RevisionTag,
+		repairStore?: ReadonlyRepairDataStore,
+	): Iterable<Delta.Root> {
```

Comment:
This is emitting a delta for each change rather than a single squashed delta because composing the changes causes a rollback test to fail. It seems that this is likely a bug in compose. If/when that is fixed, this can emit a single delta to be more efficient. Will add a TODO comment after I've finished investigating with Yann.